nussl/separation/nmf_mfcc.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import sklearn.cluster
+import librosa
+import nussl.audio_signal
+import nussl.constants
+import nussl.spectral_utils
+import nussl.utils
+from nussl.transformers import transformer_nmf
+from nussl.separation import separation_base
+import mask_separation_base
+import masks
+
+
+class NMF_MFCC(mask_separation_base.MaskSeparationBase):
+    """
+        Non Negative Matrix Factorization using K-Means Clustering on MFCC (NMF MFCC) is a source separation
+        algorithm that runs Transformer NMF on the magnitude spectrogram of an input audio signal.
+        It uses K means clustering to cluster the templates and activations returned by the NMF. The dot product of the
+        clustered templates and activations results in a magnitude spectrogram only containing a separated source.
+        This is used to create a Binary Mask object, and the whole process can be applied for each cluster to
+        return a list of Audio Signal objects corresponding to each separated source.
+
+        References:
+            Mel Frequency Cepstral Coefficients(MFCC): https://en.wikipedia.org/wiki/Mel-frequency_cepstrum
+            Volker Gnann, Martin Spiertz. "Source-filter based clustering for monaural blind source separation"
+            October 2002: https://www.researchgate.net/profile/Martin_Spiertz/publication/228567105_Source-filter_based
+            _clustering_for_monaural_blind_source_separation/links/0c960526b56dae0c69000000.pdf
+            scikit-learn KMeans: http://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html
+
+        Parameters:
+            input_audio_signal (np.array): An Audio Signal object of the input audio signal.
+            num_sources (int): Number of sources to find.
+            num_templates (int): Number of template vectors to used in NMF.
+            distance_measure (str): The type of distance measure to use in NMF - euclidean or divergence.
+            num_iterations (int): The number of iterations to go through in NMF.
+            random_seed (int): The seed to use in the numpy random generator
+            kmeans_kwargs (dict): The kwargs for KMeans parameters.
+            convert_to_mono (bool): Given a stereo signal, convert to mono.
+            mask_type (object): A soft or binary mask object used for the source separation.
+            mfcc_range (int/list/tuple): The range of mfcc to use for clustering.
```

Comment:
Use commas instead of slashes. Also this should describe how to use this nontrivial parameter. I don't see a description anywhere. Don't make people guess.